### PR TITLE
Travis improve

### PR DESCRIPTION
**Pull request by Soullivaneuh**
This PR is a Travis improvement suggestion.

With this new config, we have:

 * Tests from php 5.3 to 5.6
 * Tests form SF 2.3 to 2.6 and 2.7/2.8/3.0 allowed failure (in order to prevent deprecated issues)
 * Cache system to make tests quicker

Don't hesitate to tell me your thinking about this suggestion! :+1: 

Thanks

### CODE REVIEW

**Comment by willdurand** [comment]
:+1:

**Comment by Soullivaneuh** [comment]
This reveal failing test with lowest dependencies.

Will investigate. Any idea of why?

**Comment by Soullivaneuh** [comment]
Seems to be related to `willdurand/propel-typehintable-behavior` package with a to low propel deps.

Will try this package alone to confirm it.

**Comment by Soullivaneuh** [comment]
@willdurand can you please take a look to https://github.com/willdurand/TypehintableBehavior/pull/13 ?

Thanks.

File: .travis.yml [hunk 1]
```diff
@@ -5,9 +5,47 @@ php:
   - 5.4
   - 5.5
   - 5.6
+  - hhvm
```
Comment (Spomky) [inline .travis.yml]: - 7

Comment (Soullivaneuh) [inline .travis.yml]: PHP 7 is not stable yet. It's tested by the nightly build.

**Comment by Soullivaneuh** [comment]
@willdurand Please restart Travis jobs after pushing a new patch version on https://github.com/willdurand/TypehintableBehavior

**Comment by stof** [comment]
@Soullivaneuh all my comments on https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1824 are valid here too

**Comment by Soullivaneuh** [comment]
@stof done. Waiting for @willdurand request.

**Comment by willdurand** [comment]
thank you @Soullivaneuh

**Pull request merged**